src/components/Carousel.vue

```diff
@@ -0,0 +1,276 @@
+<template>
+    <div class="carousel" :class="{ disabled }">
+        <div v-for="entry in entries" :ref="entry"
+            :class="{ selected: effectiveSelected === entry }"
+            @click="!disabled && _updateSelection(entry)"
+            @focusin="!disabled && _updateSelection(entry)">
+            <slot :name="entry"></slot>
+        </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Watch, Vue } from 'vue-property-decorator';
+import SmallPage from './SmallPage.vue';
+import { Tweenable } from '@nimiq/utils';
+
+@Component({components: {SmallPage}})
+export default class Carousel extends Vue {
+    @Prop({
+        type: Array,
+        default: () => [],
+        validator: (entries: any) => Array.isArray(entries)
+            && entries.length > 0
+            && !entries.some((entry) => typeof entry !== 'string'),
+    })
+    public entries!: string[];
+
+    @Prop(String)
+    public selected?: string;
+
+    @Prop({
+        type: Number,
+        default: 16,
+    })
+    public entryMargin!: number;
+
+    @Prop({
+        type: Number,
+        default: 1000,
+    })
+    public animationDuration!: number; // in ms
+
+    @Prop({
+        type: Boolean,
+        default: false,
+    })
+    public hideBackgroundEntries!: boolean;
+
+    @Prop({
+        type: Boolean,
+        default: false,
+    })
+    public disabled!: boolean;
+
+    public $refs: { [ref: string]: HTMLElement[] }; // these are arrays because of v-for
+
+    private effectiveSelected: string = '';
+    private radius: Tweenable = new Tweenable();
+    private rotations: Map<string, Tweenable> = new Map(); // map entry -> rotation
+    private requestAnimationFrameId: number | null = null;
+
+    private get _hasDummyPosition(): boolean {
+        // add dummy to avoid that second entry is hidden exactly behind selected item on opposite side of circle.
+        return this.entries.length <= 2;
```

Comment:
Maybe a dummy could be added whenever the number of entries is even? Because I think it also makes sense to add for 4 elements, and for 6, and so on?

```suggestion
        return this.entries.length % 2 === 0;
```

Reply:
For >3 elements it looks/behaves better without adding a dummy.